xarray/core/computation.py

```diff
@@ -40,6 +40,12 @@
 _JOINS_WITHOUT_FILL_VALUES = frozenset({"inner", "exact"})
 
 
+def _first_of_type(args, kind):
```

Comment:
Can it happen that none is found?

Reply:
should not be possible. I'm raising `NotImplementedError` now